validator/src/admin_rpc_service.rs

```diff
@@ -212,6 +216,16 @@ pub trait AdminRpc {
         meta: Self::Metadata,
         pubkey_str: String,
     ) -> Result<HashMap<RpcAccountIndex, usize>>;
+
+    // This method does not itself do an accounts scan, but returns data only relevant to nodes
+    // that do support accounts-scan RPC apis
+    #[rpc(meta, name = "getLargestIndexKeys")]
+    fn get_largest_index_keys(
+        &self,
+        meta: Self::Metadata,
+        secondary_index: RpcAccountIndex,
+        max_entries: usize,
+    ) -> Result<RpcResponse<Vec<(String, usize)>>>;
```

Comment:
I removed RpcResponse from `get_secondary_index_key_size` when I moved it to admin_rpc_service, my rationale being that these endpoints are to give a general picture of the secondary indexes and the exact slot in time is not super important. Wdyt? If in agreement, please make the return types consistent.

Reply:
Also, see recommendation below to convert Vec to HashMap

Reply:
Not very opinionated on this. I tend to agree considering I don't know how the typical user would use this call besides on startup to filter accounts. In that context, exact slot doesn't seem too important. I must have missed the type removal on the other RPC call or I would have done the same here.